### ISSUE TITLE
pSynclient & ~/.flsynclient

### ISSUE THREAD

**Issue by wdlkmpx** [puppylinux-woof-CE/woof-CE#1660]
@rizalmart 

- pSynclient should be able to read ~/.flsynclient
- flsynclient should be a symlink to psynclient

psynclient should understand: -s or --set ($1) = `psynclient -l`
- silently apply $HOME/.config/psynclient.conf or $HOME/.flsynclient

Code to detect if the synaptics driver is loaded
```
#check if synaptics driver is loaded
TOUCHPAD_SETTINGS="$(synclient -l 2>&1)"
if [ $? -ne 0 ] ; then
	echo "$TOUCHPAD_SETTINGS" #errmsg
	exit 1
fi
```

Commit https://github.com/puppylinux-woof-CE/petbuilds/commit/33ecb7354aeef6b4f5f86536c8c1c0558ce7c8bf deletes the flsynclient and fltk petbuilds

Commit https://github.com/puppylinux-woof-CE/woof-CE/commit/f5b78565ebe0fa6abf815230a5b891637c8d1cd5 enforces psynclient

So I guess a comp

**Comment by rizalmart** [comment]
It is now implemented on psynclient version 1.3.2. The package was already uploaded

**Comment by peabee** [comment]
flSynclient provides useful pop-up help when you hover over each control option.... can psynclient provide equivalent help??
![Screenshot](https://user-images.githubusercontent.com/5926865/68471940-3a124280-0217-11ea-8f56-2a9dda625f05.png)

**Comment by wdlkmpx** [comment]
gtkdialog stuff is advanced but maybe not so advanced.

My advice:

- use flsynclient and copy the help messages, if you can't create
'tooltips', just add a Help button and text (see pupswap, quicksetup).
- take your time to test and receive feedback

Doing this stuff at this point seems a bit masochistic, that's why
flsynclient had to be removed. It had to be removed there was no
choice.

I'll keep this open for a few weeks.

**Comment by zigbert** [comment]
<hbox tooltip-text=" This is the text ">...</hbox>

**Comment by zigbert** [comment]
`<hbox tooltip-text=" This is the text ">...</hbox>`

**Comment by zigbert** [comment]
Add tooltips with advanced text layout.
`<button tooltip-markup="<span background='yellow'><b><i> press alt m </i><b><span>">......</button>`

**Comment by radky2** [comment]
@rizalmart,

Thanks for this great application!

---

Re: Testing psynclient 1.3.2

Currently, the following option has no effect on my touchpad:

Tapping tab -> Drag tab -> Locked Drag Timeout

However, this option works correctly if I modify the psynclient xSETTINGS list,

and change:

LockedDragTimeout=$(boolnum $LOCKED_DRAG_TIMEOUT)

to

LockedDragTimeout=$LOCKED_DRAG_TIMEOUT

---

Also, it seems the following options do not apply user input (at least, not for my Acer touchpad when using the Debian synaptics input driver of BusterPup):

1 - General tab -> Disable Guest Mouse
2 - General tab -> Sensitivity -> Press
3 - Movement tab -> Track Stick Speed
4 - Tapping tab -> Timing tab -> Fast Taps
5 - Scrolling tab -> Edge tab -> Edge Motion
6 - Scrolling tab -> Edge tab -> Min Edge Speed
7 - Scrolling tab -> Edge tab -> Max Edge Speed

This observation may be explained by the following link which indicates the above seven options are among those parameters that are discontinued and no longer part of the Debian synaptics driver configuration:

https://manpages.debian.org/jessie/xserver-xorg-input-synaptics/synaptics.4.en.html

REMOVED OPTIONS:
1 - GuestMouseOff
2 - FingerPress
3 - TrackstickSpeed
4 - FastTaps
5 - EdgeMotionUseAlways
6 - EdgeMotionMinSpeed
7 - EdgeMotionMaxSpeed

Perhaps these options are still functional for those who use old versions of the synaptics input driver.

**EDIT:** Maybe others could test if the same issues pertain to non-Debian synaptic input drivers.

**Comment by rizalmart** [comment]
@peabee tooltips has been added to version 1.4. Synaptics help guide was came from bsd, arch, gentoo, and ubuntu wikis and manpages
@radky2 The bug is now fixed on version 1.4. I intentionally added depreciated synaptics settings for older version of synaptic driver

**Comment by wdlkmpx** [comment]
This seems to be a generic package and an ambitious project, although the code needs some proper indenting.

I haven't tested this of course.  It probably requires a certain level of collaboration that is normally found in git repos.

But in puppy this is how it works:
1. You post a release pet package.
2. People test it and provide feedback and suggest code changes or upload the whole script with the changes (no diffs)
3. You fix the issues or apply the suggested changes
4. You  release a new package
5. New issues are discovered as a result of the changes (probably a syntax error)
6. You fix the issues and release a new tarball
7. People continue testing and find new issues 
8. You fix the issues and release a new tarball
9. After 2 or 3 releases in the forum, the package is probably ready to be uploaded to noarch.
